### PR TITLE
array_merge_recursive rather than delay setting the flash messages cookie

### LINKED ISSUE #558: Empty flash_messages cookie in tests with v2.9.1
Hi,

when upgrading to version 2.9.1 my tests fail due to empty cookie jar of flash messages in tests.
With version 2.9.0 everything works just fine.
I have seen that version 2.9.1 should fix something with multiple redirects and flash messages, so I guess this has something to do with that.
Do I need to make any changes in my code / config?

### PR DESCRIPTION
**Pull request by dbu**
fix #558

### CODE REVIEW

**Comment by dbu** [comment]
@ismailcherri @spointecker may I ask you two to review this pull request and tell if you see any new pitfalls i might be introducing here?

File: src/EventListener/FlashMessageListener.php [hunk 1]
```diff
@@ -87,26 +87,28 @@ public function onKernelResponse(FlashMessageResponseEvent $event)
...
         if (isset($cookies[$host][$this->options['path']][$this->options['name']])) {
             $rawCookie = $cookies[$host][$this->options['path']][$this->options['name']]->getValue();
-            $flashes = array_merge($flashes, json_decode($rawCookie));
+            $flashes = array_merge_recursive($flashes, json_decode($rawCookie, true));
+        }
+
+        // Preserve existing flash message cookie from previous redirect if there was one.
+        // This covers multiple redirects where each redirect adds flash messages.
+        $request = $event->getRequest();
+        if ($request->cookies->has($this->options['name'])) {
```
Comment (dbu) [inline src/EventListener/FlashMessageListener.php:109]: before, we did not look at request cookies at all.

i was wondering if there is a security risk involved with this (depending on what weird things a dev might put into a flash message) but i guess if the attacker manages to inject a cookie for this domain, they can also do more direct attacks with javascript that do not involve looping through the backend.

**Comment by spointecker** [comment]
LGTM. Thanks for the quick answer and fix, will test it with my app&tests as soon as it is released!

**Comment by ismailcherri** [comment]
LGTM. Thanks for the fix.

**Comment by dbu** [comment]
thanks for the quick feedback!

released https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/releases/tag/2.10.2 and https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/releases/tag/2.9.2

**Pull request merged**